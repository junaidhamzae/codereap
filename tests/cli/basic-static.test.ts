import path from 'node:path';
import fs from 'node:fs';
import os from 'node:os';
import { runCli } from '../helpers/cli';
import { normalizeReport } from '../helpers/normalize';

describe('CLI basic-static fixture', () => {
  it('finds D.ts as orphan and outputs required keys', () => {
    const fixture = path.resolve(__dirname, '../fixtures/basic-static');
    const tmp = fs.mkdtempSync(path.join(os.tmpdir(), 'cli-'));
    const outBase = path.join(tmp, 'report');
    const { stdout, status, json } = runCli([
      '--root', fixture,
      '--out', outBase,
    ]);
    expect(status).toBe(0);
    expect(Array.isArray(json)).toBe(true);
    const rows = normalizeReport(json as any[]);
    const dRow = rows.find(r => r.node.endsWith('D.ts'))!;
    expect(dRow.orphan).toBe(true);
    expect(rows).toMatchSnapshot();
    // sanity keys
    expect(dRow).toEqual(expect.objectContaining({ node: expect.any(String), exists: true, orphan: expect.any(Boolean) }));
  });
});


