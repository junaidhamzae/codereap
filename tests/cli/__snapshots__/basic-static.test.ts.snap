// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CLI basic-static fixture finds D.ts as orphan and outputs required keys 1`] = `
[
  {
    "exists": true,
    "in-degree": 0,
    "node": "A.ts",
    "orphan": false,
    "size-bytes": 16,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
    },
  },
  {
    "exists": true,
    "in-degree": 1,
    "node": "B.ts",
    "orphan": false,
    "size-bytes": 39,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "used",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
    },
  },
  {
    "exists": true,
    "in-degree": 1,
    "node": "C.ts",
    "orphan": false,
    "size-bytes": 27,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "value",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "D.ts",
    "orphan": true,
    "size-bytes": 45,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [
          {
            "name": "orphan",
            "orphan": true,
            "reexport": false,
            "referencedInFile": false,
          },
        ],
        "reExports": [],
      },
      "imports": [],
    },
  },
  {
    "exists": true,
    "in-degree": 0,
    "node": "package.json",
    "orphan": true,
    "size-bytes": 106,
    "symbols": {
      "exports": {
        "default": {
          "exists": false,
          "orphan": true,
          "referencedInFile": false,
        },
        "named": [],
        "reExports": [],
      },
      "imports": [],
    },
  },
]
`;
